File: exercise/week04/java/src/main/java/construct/NormalArtefactUpdater.java
package construct;

public class NormalArtefactUpdater implements ArtefactUpdater {
    
    @Override
    public void update(Artefact artefact) {
        decrease(artefact, 1);
        artefact.timeToLive--;
        if (artefact.timeToLive < 0) {
            decrease(artefact, 1);
        }
    }

    protected static void decrease(Artefact artefact, int amount) {
        artefact.integrity = Math.max(0, artefact.integrity - amount);
    }

    protected static void increase(Artefact artefact, int amount) {
        artefact.integrity = Math.min(50, artefact.integrity + amount);
    }
}

File: exercise/week04/php/src/Construct/NormalArtefactUpdater.php
<?php

namespace Construct;

class NormalArtefactUpdater implements ArtefactUpdater
{
    public function update(Artefact $artefact): void
    {
        $this->decrease($artefact, 1);
        $artefact->timeToLive--;
        if ($artefact->timeToLive < 0) {
            $this->decrease($artefact, 1);
        }
    }

    protected function decrease(Artefact $artefact, int $amount): void
    {
        $artefact->integrity = max(0, $artefact->integrity - $amount);
    }

    protected function increase(Artefact $artefact, int $amount): void
    {
        $artefact->integrity = min(50, $artefact->integrity + $amount);
    }
}